dali/dfu/dfurun.cpp

```diff
@@ -533,6 +533,24 @@ class CDFUengine: public CInterface, implements IDFUengine
         }
     }
 
+    void ensureAFilePermissions(IFileDescriptor * fd, SecAccessFlags perm, bool write)
+    {
+        StringBuffer traceName;
+        fd->getTraceName(traceName);
+        if (traceName.length() > 255)
```

Comment:
in fact, there's an existing function for this kind of truncation, see jstring's elideString function.